docker/simulation/carla_ros_bridge/carla_ros_bridge.Dockerfile

```diff
@@ -0,0 +1,113 @@
+ARG BASE_IMAGE=ghcr.io/watonomous/wato_monorepo/base:foxy-ubuntu20.04
+
+ARG CARLA_VERSION=0.9.13
+FROM carlasim/carla:${CARLA_VERSION} AS wato_carla_api
+
+################################ Source ################################
+FROM ${BASE_IMAGE} as source
+
+WORKDIR ${AMENT_WS}/src
+
+# Copy in source code 
+COPY src/simulation/carla_config carla_config
+COPY src/wato_msgs/simulation ros_msgs
+
+# Scan for rosdeps
+RUN apt-get -qq update && rosdep update --rosdistro foxy && \
+    rosdep install --from-paths . -r -s \
+        | grep 'apt-get install' \
+        | awk '{print $3}' \
+        | sort  > /tmp/colcon_install_list
+
+################################# Dependencies ################################
+FROM ${BASE_IMAGE} as dependencies
+
+# Install Rosdep requirements
+COPY --from=source /tmp/colcon_install_list /tmp/colcon_install_list
+RUN apt-fast install -qq -y --no-install-recommends $(cat /tmp/colcon_install_list)
+
+# Copy in source code from source stage
+WORKDIR ${AMENT_WS}
+COPY --from=source ${AMENT_WS}/src src
+
+# Dependency Cleanup
+WORKDIR /
+RUN apt-get -qq autoremove -y && apt-get -qq autoclean && apt-get -qq clean && \
+    rm -rf /root/* /root/.ros /tmp/* /var/lib/apt/lists/* /usr/share/doc/*
+
+################################ Build ################################
+FROM dependencies as build
+
+ARG CARLA_VERSION
+
+# Install dependencies
+RUN apt-get update && \
+      apt-fast install -qq -y --no-install-recommends lsb-release \
+      libglu1-mesa-dev xorg-dev \
+      software-properties-common \
+      build-essential \
+      wget \
+      curl \
```

Comment:
if you can, either move these to the source stage (line 8) or remove. Security risk.